pkg/healthcheck/healthcheck.go

```diff
@@ -2113,7 +2113,7 @@ func (hc *HealthChecker) getDataPlanePods(ctx context.Context) ([]*pb.Pod, error
 	resp, err := hc.apiClient.ListPods(ctx, req)
 	if err != nil {
 		// return a SkipError if there is no prometheus present
-		if strings.HasSuffix(err.Error(), "No prometheus instance to connect") {
+		if strings.Contains(err.Error(), public.ErrNoPrometheusInstance) {
```

Comment:
After the the recent change with using `errors.Is` when possible, any reason we shouldn't do the same here?

If `ErrNoPrometheusInstance` was itself an error, we should be able to check for that here instead of using `strings.Contains`.